plugins/module_utils/vm.py

```diff
@@ -358,6 +358,19 @@ def get_by_name(
         vm_from_hypercore = cls.from_hypercore(hypercore_dict, rest_client)
         return vm_from_hypercore
 
+    @classmethod
+    def get_by_old_or_new_name(cls, ansible_dict, rest_client):
+        vm_old_name = VM.get_by_name(ansible_dict, rest_client)
+        if ansible_dict["vm_name_new"] is None:
```

Comment:
`vm_name_new = ansible_dict["vm_name_new"]`
Can probably remove this `if ansible_dict["vm_name_new"] is None:` and just do the `else`.
The code `vm = vm_old_name or vm_new_name` already checks for `None`.

Reply:
You are right, this wil be better, thank you. And VM.get_by_name does return None if nothing is found, so the 
`vm_name_new = VM.get_by_name() if ... else None` reduces to just `vm_name_new = VM.get_by_name()`.